src/ioq_sup.erl

```diff
@@ -40,8 +42,10 @@ init([]) ->
     }}.
 
 ioq_server2_children() ->
+    Name = ?IOQ2_SEARCH_SERVER,
+    Search = {Name, {ioq_server2, start_link, [Name, search, false]}, permanent, 5000, worker, [Name]},
```

Comment:
I've noticed that you are using `re:` module to filter the processes you need. I have a suggestion how to improve that. In erlang's supervisor the child id is not necessary an atom, it is any term. Which means you can encode the type of a child (or scheduler number) into the child id. In this case you can switch from using strings to using atoms. Matching on atoms would speed up the filter_children function a lot.


```


ioq_server2_children() ->
    Name = ?IOQ2_SEARCH_SERVER,
    Search = {{ioq2, Name}, {ioq_server2, start_link, [Name, search, false]}, permanent, 5000, worker, [Name]},
    Bind = config:get_boolean("ioq2", "bind_to_schedulers", false),
    [Search | ioq_server2_children(erlang:system_info(schedulers), Bind)].

ioq_server2_children(Count, Bind) ->
    lists:map(fun(I) ->
        Name = list_to_atom("ioq_server_" ++ integer_to_list(I)),
        {{ioq2, Name}, {ioq_server2, start_link, [Name, I, Bind]}, permanent, 5000, worker, [Name]}
    end, lists:seq(1, Count)).

processes(ioq2) ->
    filter_children(ioq2);
processes(ioq) ->
    filter_children(ioq_server);
processes(config_listener_mon) ->
    filter_children(config_listener_mon);
processes(Arg) ->
    {error, [
        {expected_one_of, [ioq, ioq2, config_listener_mon]},
        {got, Arg}]}.

filter_children(Kind) ->
    lists:filtermap(fun({ChildId, P, _, _}) ->
        case ChildId of
            Kind -> {true, {ChildId, P}};
            {Kind, Id} -> {true, {Id, P}}
            _ -> false
        end
    end, supervisor:which_children(?MODULE)).
```

Reply:
I like this idea, but it would break the current expectation that registered processes are atom names that can be converted into a string key of a json dictionary when we generate the `/_system` status endpoint; perhaps other places in the codebase too. For example:

```
(node1@127.0.0.1)1> jiffy:encode({[{ioq_server1, foo}]}).
<<"{\"ioq_server1\":\"foo\"}">>
(node1@127.0.0.1)2> jiffy:encode({[{{ioq2, ioq_server1}, foo}]}).
** exception error: {invalid_object_member_key,{ioq2,ioq_server1}}
     in function  jiffy:encode/2 (src/jiffy.erl, line 99)
```